### PR TITLE
SCC: Support for bounds aliases and derived type members as bounds

### PR DESCRIPTION
**Pull request by awnawab**
This is the final PR needed to transform the SL. It consists of a series of fixes to the SCC transforms, the biggest of which is the ability to use aliased components of a derived-type as the horizontal loop bounds.

NB: this is stacked on top of #249.

### CODE REVIEW

**Comment by awnawab** [comment]
Can I push a small fix to the dependency transform here or shall I file a separate PR for that?

**Comment by reuterbal** [comment]
> Can I push a small fix to the dependency transform here or shall I file a separate PR for that?

I would hold off or make it a separate PR. That transforms has been reshuffled a bit on the new Scheduler branch.

**Comment by awnawab** [comment]
@reuterbal I have rebased and fixed this PR and #249, please let me know when I can push changes to both.

**Comment by reuterbal** [comment]
> @reuterbal I have rebased and fixed this PR and #249, please let me know when I can push changes to both.

I've merged the backlog of already reviewed PRs now, so now is likely a good point to rebase your PRs. Many thanks!

**Comment by awnawab** [comment]
This has a conflict now with #270, so I am turning this into a draft until that is reviewed and merged.

**Review by reuterbal — CHANGES_REQUESTED** [review]
Thanks, this looks great!

I've left a few comments but no fundamental show stoppers.

**Review by mlange05 — COMMENTED** [review]
I agree with what @reuterbal pointed out, and have nothing further to add. I can confirm that this runs fine with ec-physics regression.

File: loki/tools/util.py
Comment (awnawab) [reply loki/tools/util.py:814]: Ah yes, that would be much cleaner, thanks! I assume you mean `ProgramUnit` though rather than `Scope`, because that is where the `variables` and `variable_map` are computed?

Comment (reuterbal) [reply loki/tools/util.py:814]: Yes, thanks, `ProgramUnit` would be the right place.

File: transformations/transformations/single_column_base.py
Comment (awnawab) [reply transformations/transformations/single_column_base.py:112]: Many many thanks for this 🙏 this is much cleaner and simpler than what I original had 👌

File: transformations/transformations/single_column_base.py
Comment (awnawab) [reply transformations/transformations/single_column_base.py:296]: Yes I agree with the name change, but I'll keep the routine as an argument because I would like to also print the routine name to have a more informative error message.

**Review by reuterbal — APPROVED** [review]
Many thanks, very happy with this now!

**Pull request merged**
